### PR TITLE
Fix #253

### LINKED ISSUE #253: [PE-D][Tester D] Result display does not match shown list
I had a trainer selected (Bernice Yu) with one client on the right. The client list had a button showing "Showing: Bernice Yu". I tried entering the command `find David` because I wanted to find clients with name David. The result display showed "1 persons listed!" but the UI did not show any persons.

The issue was that client list only shows clients under trainer Bernice Yu. In the UG, the `find` command mentions that it finds trainers and clients as long as their names match any keyword. As such, the expected UI should show David in the client list with the button labelled "Showing: Filtered".

Actual UI:

Image

From this page, I clicked "Showing: Bernice Yu" and the button turned to "Show All Clients" with all the clients being listed.

Therefore, the `find` command does not work as intended if the right button is "Showing: <trainer name>" before the command is executed.

-----------------------------------------------------------------------------
Steps to reproduce (running on Windows 11):

1. Navigate to the folder containing GymOps.jar on File Explorer.
2. Inside the folder, right-click and select "Open in Terminal".
3. Run the command `java -jar GymOps.jar` on the terminal window.
4. The app launches.
5. Click on trainer Bernice Yu. The right button now shows "Showing: Bernice Yu"
6. Enter the command `find david`. The result display mentions "1 persons listed!" but both lists are empty. The right button still shows "Showing: Bernice Yu"

This issue is problematic if the user accidentally clicks a trainer before they enter a `find` command.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`

### PR DESCRIPTION
**Pull request by ephtale**

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/FindCommandTest.java [hunk 1]
```diff
@@ -92,9 +92,8 @@ public void execute_preservesSelectedTrainer() throws Exception {
         // Run find with a keyword that matches nothing
         new FindCommand(preparePredicate("Nonexistent")).execute(model);
 
-        // Trainer selection must still be present
-        assertTrue(model.getSelectedTrainer().isPresent());
-        assertEquals(trainer, model.getSelectedTrainer().get());
+        // Trainer selection must be cleared so client list can show filtered results
+        assertTrue(model.getSelectedTrainer().isEmpty());
```
Comment (TheSputnikSpacecraft) [inline src/test/java/seedu/address/logic/commands/FindCommandTest.java:96]: Looks Good, Nice fix to #253 , i was wondering how we would tackle that

**Review by TheSputnikSpacecraft — APPROVED** [review]
